tmt/base.py

```diff
@@ -542,16 +542,18 @@ def export(self, format_='yaml', keys=None):
             tmt.export.export_to_nitrate(self)
 
         # Common node export otherwise
-        elif keys == 'fmf-id':
-            if format_ == 'dict':
+        elif format_ == 'dict' or format_ == "yaml":
```

Comment:
Current change causes that `Test.export()` returns a `dict` when asked for `yaml`. For example:

    >>> test = tmt.Tree('.').tests()[0]
    >>> test.export(format_='yaml')
    {'name': '/tests/clean/guests', 'summary': 'Checks whether tmt clean...

This is not expected and could be very confusing behaviour.

Reply:
How about `Plan.export()` return if we asked for `yaml`? e.g.
```
>>> plan = tmt.Tree('.').plans()[0]
>>> plan.export(format_='yaml')
```

Reply:
With patch in the following,
```patch
diff --git a/tmt/base.py b/tmt/base.py
index 5457b10..cba5594 100644
--- a/tmt/base.py
+++ b/tmt/base.py
@@ -550,7 +550,10 @@ class Test(Core):
                 data['name'] = self.name
                 data.update(super().export(format_='dict'))
                 data.pop('adjust', None)
-                return data
+                if format_ == 'dict':
+                    return data
+                else:
+                    return tmt.utils.dict_to_yaml(data)
         else:
             raise tmt.utils.GeneralError(
                 f"Invalid test export format '{format_}'.")
diff --git a/tmt/cli.py b/tmt/cli.py
index e8875f5..3393624 100644
--- a/tmt/cli.py
+++ b/tmt/cli.py
@@ -581,15 +581,12 @@ def export(context, format_, nitrate, bugzilla, **kwargs):
         if kwargs.get('fmf_id'):
             keys = 'fmf-id'
 
-        data = list()
-        for test in context.obj.tree.tests():
-            test_info = test.export(format_=format_, keys=keys)
-            data.append(test_info)
-
+        tests = [test.export(format_='dict', keys=keys) for test in
+            context.obj.tree.tests()]
         if format_ == 'dict':
-            echo(data, nl=False)
+            echo(tests, nl=False)
         else:
-            echo(tmt.utils.dict_to_yaml(data), nl=False)
+            echo(tmt.utils.dict_to_yaml(tests), nl=False)
     else:
         raise tmt.utils.GeneralError(
             f"Invalid test export format '{format_}'.")
```

`test.export(format_='yaml')` works fine now. 
```python
>>> test = tmt.Tree('.').tests()[0]
>>> test.export(format_='yaml')
'name: /tests/example1\nsummary: Test01\ndescription:\ncontact: []\ncomponent: []\ntest: ./test.sh\npath: /tests/example1\nframework: beakerlib\nmanual: false\nrequire:\n  - beakerlib\nrecommend: []\nenvironment: {}\nduration: 5m\nenabled: true\norder: 50\nresult: respect\ntag: []\ntier:\nlink: []\n'
>>>
>>> test.export(format_='dict')
{'name': '/tests/example1', 'summary': 'Test01', 'description': None, 'contact': [], 'component': [], 'test': './test.sh', 'path': '/tests/example1', 'framework': 'beakerlib', 'manual': False, 'require': ['beakerlib'], 'recommend': [], 'environment': {}, 'duration': '5m', 'enabled': True, 'order': 50, 'result': 'respect', 'tag': [], 'tier': None, 'link': []}
>>> 
>>> 
>>> plan = tmt.Tree('.').plans()[0]
>>> plan.export(format_='yaml')
'name: /plans/example1\nsummary: Basic smoke test 01\ndescription:\nenabled: true\norder: 50\nlink: []\ndiscover:\n    how: fmf\n    name: default\nexecute:\n    how: tmt\n    name: default\n'
>>>
>>> plan.export(format_='dict')
{'name': '/plans/example1', 'summary': 'Basic smoke test 01', 'description': None, 'enabled': True, 'order': 50, 'link': [], 'discover': {'how': 'fmf', 'name': 'default'}, 'execute': {'how': 'tmt', 'name': 'default'}}
```

Reply:
Yes, now works as expected. No need to include patches in the comments. Just add the fix as a separate commit. If needed, you can then easily reference it using the commit hash.